internal/provider/aws/imds.go

```diff
@@ -0,0 +1,116 @@
+package aws
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"sync"
+	"time"
+)
+
+const (
+	// Retrieve instance metadata for AWS EC2 instance
+	// https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html
+	instanceMetadataEndpoint = "http://169.254.169.254/latest/meta-data"
+
+	// IMDSv2 token related constants
+	tokenEndpoint      = "http://169.254.169.254/latest/api/token"
+	tokenTTLHeader     = "X-aws-ec2-metadata-token-ttl-seconds"
+	tokenRequestHeader = "X-aws-ec2-metadata-token"
+)
+
+// The VPC identifier
+// Automatically retrieved with GetVPCID function.
+// For run outside of the cluster, can be set through linker flag, e.g.
+// go build -ldflags "-X github.com/quortex/kubestatic/internal/provider/aws.vpcID=$VPC_ID" -a -o manager main.go
+var (
+	vpcID   string
+	vpcOnce sync.Once
+	vpcErr  error
```

Comment:
I don't think the error should be set here, it only needs to be returned when the singleton is executed.